granted: 0.34.1 -> 0.35.2

jlbribeiro opened the pull request:
- **Diff:** https://github.com/common-fate/granted/compare/v0.34.1...v0.35.2
- **Changelog:**
    - https://github.com/common-fate/granted/releases/tag/v0.35.0
    - https://github.com/common-fate/granted/releases/tag/v0.35.1
    - https://github.com/common-fate/granted/releases/tag/v0.35.2
- **Maintainers:**
  - **currently:** @ivankovnatsky;
  - **after this PR:** @jlbribeiro (me), following [this comment](https://github.com/NixOS/nixpkgs/pull/347364#issuecomment-2402941356).

Also adds `updateScript` (`nix-update-script`) and `versionCheckHook`.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jlbribeiro committed:
granted: format using nixfmt

jlbribeiro committed:
granted: set updateScript

jlbribeiro committed:
granted: 0.34.1 -> 0.35.0

Diff: https://github.com/common-fate/granted/compare/v0.34.1...v0.35.0
Changelog: https://github.com/common-fate/granted/releases/tag/v0.35.0

jlbribeiro committed:
granted: add versionCheckHook

jlbribeiro committed:
granted: 0.35.0 -> 0.35.1

Diff: https://github.com/common-fate/granted/compare/v0.35.0...v0.35.1
Changelog: https://github.com/common-fate/granted/releases/tag/v0.35.1

ivankovnatsky commented on pkgs/by-name/gr/granted/package.nix:
```diff
@@ -38,23 +41,30 @@ buildGoModule rec {
-      --suffix PATH : ${lib.makeBinPath [ xdg-utils ]}
+      wrapProgram $out/bin/assume \
+        --suffix PATH : ${lib.makeBinPath [ xdg-utils ]}
 
```
```suggestion
```

ivankovnatsky reviewed (approved):
I'm sorry, I don't use it currently, but otherwise it looks good, thanks.

jlbribeiro commented:
@ivankovnatsky I currently use it at work; would you want me to replace you as a maintainer?

jlbribeiro committed:
granted: add jlbribeiro as maintainer

Removes @ivankovnatsky as maintainer following
https://github.com/NixOS/nixpkgs/pull/347364#pullrequestreview-2357233641

ivankovnatsky commented:
> @ivankovnatsky I currently use it at work; would you want me to replace you as a maintainer?
> 
> **edit:** Took the liberty of doing just that; please feel free to say you rather just have me be an additional maintainer. Because I feel maintainer changes are important I'm re-requesting your code review (hope you don't mind), and updating the PR description to reflect that.
> 
> (also addressed your whitespace suggestion in the process)

Sure thing, totally makes sense! Thanks.

pjrm commented:
I've tested on aarch64-darwin and  x86_64-linux, and it works as expected.

Thanks 🙏

pbsds commented:
wow it's jarring to be be about to click the merge button, and see it turn gray due to a merge-conflict

conflict with https://github.com/NixOS/nixpkgs/pull/347816

jlbribeiro commented:
@pbsds May I kindly ask you to re-review + merge after the force-push? I've resolved the conflicts (caused by the `nixfmt` commit).
(easier to review [that commit](https://github.com/NixOS/nixpkgs/commit/a34f21576968d0d68aaed40106a3e9a7d7f5bcc3) separate from [the rest](https://github.com/NixOS/nixpkgs/compare/a34f21576968d0d68aaed40106a3e9a7d7f5bcc3..jlbribeiro:pkgs/granted)).

jlbribeiro committed:
granted: 0.35.1 -> 0.35.2

Diff: https://github.com/common-fate/granted/compare/v0.35.1...v0.35.2
Changelog: https://github.com/common-fate/granted/releases/tag/v0.35.2
